### ISSUE TITLE
code-server not loading PDFs from built-in PDF viewer

### ISSUE THREAD

**Issue by raulbalanza** [James-Yu/LaTeX-Workshop#3796]
## Pre-checks*
- [x] The issue has not been reported in this repository.
- [x] The issue remains after disabling all other extensions and restarting Visual Studio Code.
- [x] The [FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) cannot address the issue.
- [x] The issue is not related to compiling a document, or the document can be successfully compiled in the OS terminal but not in Visual Studio Code with this extension.

## Environment*
- Operating System: Ubuntu Server (through Docker)
- Visual Studio Code Version: 1.76.1
- LaTeX Workshop Version: 9.7.0
- TeX Distribution Version: TeX Live 2022
- I am using code-server whose version is v4.11.0 inside a Docker container.

## The Issue*
I cannot view PDF files in the built-in PDF viewer of LaTeX Workshop. However, I can do so by using another extensions (e.g., Papyrus PDF Preview).

### Reproduction Steps
1. Deploy code-server (I used this Docker image: https://hub.docker.com/r/linuxserver/code-server)
2. Install texlive from the command line
3. Install LaTeX Workshop
4. Try to view a PDF file (no other extensions installed)

### Expected Behavior
- I expect to see the loaded PDF file like I do on my local VSCode/LaTeX install

### LaTeX Workshop Output*
```
[18:23:45][Logger] New log placeholder %WS1% registered for /config/workspace .
[18:23:45][Config] Configuration for workspace: file://%WS1% .
[18:23:45][Config] editor.acceptSuggestionOnEnter: "on" .
[18:23:45][Config] latex-workshop.bind.enter.key: true .
[18:23:45][Config] latex-workshop.docker.enabled: false .
[18:23:45][Config] latex-workshop.docker.image.latex: "" .
[18:23:45][Config] latex-workshop.hover.preview.mathjax.extensions: [] .
[18:23:45][Config] latex-workshop.intellisense.package.enabled: true .
[18:23:45][Config] latex-workshop.intellisense.update.aggressive.enabled: false .
[18:23:45][Config] latex-workshop.intellisense.update.delay: 1000 .
[18:23:45][Config] latex-workshop.latex.autoBuild.run: "onFileChange" .
[18:23:45][Config] latex-workshop.latex.build.forceRecipeUsage: true .
[18:23:45][Config] latex-workshop.latex.outDir: "%DIR%" .
[18:23:45][Config] latex-workshop.latex.recipes: [
 {
  "name": "latexmk",
  "tools": [
   "latexmk"
  ]
 },
 {
  "name": "latexmk (latexmkrc)",
  "tools": [
   "latexmk_rconly"
  ]
 },
 {
  "name": "latexmk (lualatex)",
  "tools": [
   "lualatexmk"
  ]
 },
 {
  "name": "latexmk (xelatex)",
  "tools": [
   "xelatexmk"
  ]
 },
 {
  "name": "pdflatex -> bibtex -> pdflatex * 2",
  "tools": [
   "pdflatex",
   "bibtex",
   "pdflatex",
   "pdflatex"
  ]
 },
 {
  "name": "Compile Rnw files",
  "tools": [
   "rnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "Compile Jnw files",
  "tools": [
   "jnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "Compile Pnw files",
  "tools": [
   "pnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "tectonic",
  "tools": [
   "tectonic"
  ]
 }
] .
[18:23:45][Config] latex-workshop.latex.tools: [
 {
  "name": "latexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-pdf",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "lualatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-lualatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "xelatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-xelatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "latexmk_rconly",
  "command": "latexmk",
  "args": [
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "pdflatex",
  "command": "pdflatex",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "bibtex",
  "command": "bibtex",
  "args": [
   "%DOCFILE%"
  ],
  "env": {}
 },
 {
  "name": "rnw2tex",
  "command": "Rscript",
  "args": [
   "-e",
   "knitr::opts_knit$set(concordance = TRUE); knitr::knit('%DOCFILE_EXT%')"
  ],
  "env": {}
 },
 {
  "name": "jnw2tex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"tex\")"
  ],
  "env": {}
 },
 {
  "name": "jnw2texminted",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"texminted\")"
  ],
  "env": {}
 },
 {
  "name": "pnw2tex",
  "command": "pweave",
  "args": [
   "-f",
   "tex",
   "%DOC_EXT%"
  ],
  "env": {}
 },
 {
  "name": "pnw2texminted",
  "command": "pweave",
  "args": [
   "-f",
   "texminted",
   "%DOC_EXT%"
  ],
  "env": {}
 },
 {
  "name": "tectonic",
  "command": "tectonic",
  "args": [
   "--synctex",
   "--keep-logs",
   "%DOC%.tex"
  ],
  "env": {}
 }
] .
[18:23:45][Config] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto" .
[18:23:45][Manager] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[18:23:45][Server] Creating LaTeX Workshop http and websocket server.
[18:23:45][Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":33339} .
[18:23:45][Server] valdOrigin is https://mydomain.example.com
[18:23:45][Extension] Initializing LaTeX Workshop.
[18:23:45][Extension] Extension root: /config/extensions/james-yu.latex-workshop-9.7.0-universal
[18:23:45][Extension] $PATH: /app/code-server/lib/vscode/bin/remote-cli:/command:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
[18:23:45][Extension] $SHELL: undefined
[18:23:45][Extension] $LANG: en_US.UTF-8
[18:23:45][Extension] $LC_ALL: undefined
[18:23:45][Extension] process.platform: linux
[18:23:45][Extension] process.arch: x64
[18:23:45][Extension] vscode.env.appName: code-server
[18:23:45][Extension] vscode.env.remoteName: mydomain.example.com
[18:23:45][Extension] vscode.env.uiKind: 2
[18:23:45][Extension] LaTeX Workshop initialized.
[18:23:45][Format][Bib] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[18:23:45][Extension] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[18:23:45][Manager] Current workspace folders: ["file://%WS1%"]
[18:23:45][Manager] Current workspaceRootDir: file://%WS1%
[18:23:45][Cacher][Path] Non-existent .fls for %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Cache not finished on %WS1%/pruebatex/main.tex when getting its children.
[18:23:45][Cacher] Caching %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Updated inputs of %WS1%/pruebatex/main.tex .
[18:23:45][DupLabel] Checking for duplicate labels: %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Updated elements of %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Updated bibs of %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Cached %WS1%/pruebatex/main.tex .
[18:23:45][Structure] Structure cleared on undefined root.
[18:23:45][Manager] Found files that might be root, choose the first one: %WS1%/pruebatex/main.tex
[18:23:45][Manager] Root file changed: from undefined to %WS1%/pruebatex/main.tex
[18:23:45][Manager] Start to find all dependencies.
[18:23:45][Manager] Root file languageId: latex
[18:23:45][Cacher][Watcher] Reset.
[18:23:45][Cacher] Adding %WS1%/pruebatex/main.tex .
[18:23:45][Cacher][Watcher] Watched %WS1%/pruebatex/main.tex with a new watcher on %WS1%/pruebatex .
[18:23:45][Cacher] Caching %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Updated inputs of %WS1%/pruebatex/main.tex .
[18:23:45][DupLabel] Checking for duplicate labels: %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Updated elements of %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Updated bibs of %WS1%/pruebatex/main.tex .
[18:23:45][Cacher] Cached %WS1%/pruebatex/main.tex .
[18:23:45][Cacher][Path] Non-existent .fls for %WS1%/pruebatex/main.tex .
[18:23:45][Structure] Structure force updated with 1 root sections for %WS1%/pruebatex/main.tex .
[18:23:50][Viewer][Panel] Internal PDF viewer at https://mydomain.example.com/proxy/33339/viewer.html?file=pdf..ZmlsZSUzQSUyRiUyRiUyRmNvbmZpZyUyRndvcmtzcGFjZSUyRnBydWViYXRleCUyRm1haW4ucGRm .
[18:23:50][Cacher][Watcher] Watched %WS1%/pruebatex/main.pdf with a new watcher on %WS1%/pruebatex .
[18:23:50][Viewer] Open PDF tab for file://%WS1%/pruebatex/main.pdf
[18:23:51][Viewer] Handle data type: open
[18:23:59][Manager] Current workspace folders: ["file://%WS1%"]
[18:23:59][Manager] Found root file from active editor: %WS1%/pruebatex/main.tex
[18:23:59][Manager] Keep using the same root file: %WS1%/pruebatex/main.tex
[18:23:59][Structure] Structure updated with 1 root sections for %WS1%/pruebatex/main.tex .
[18:24:01][Extension] onDidSaveTextDocument triggered: file://%WS1%/pruebatex/main.tex
[18:24:01][Cacher][Watcher] "change" emitted on %WS1%/pruebatex/main.tex .
[18:24:01][Cacher] Caching %WS1%/pruebatex/main.tex .
[18:24:01][Cacher] Updated inputs of %WS1%/pruebatex/main.tex .
[18:24:01][Builder] Auto build started detecting the change of a file: %WS1%/pruebatex/main.tex .
[18:24:01][Commander] BUILD command invoked.
[18:24:01][Commander] The document of the active editor: file://%WS1%/pruebatex/main.tex
[18:24:01][Commander] The languageId of the document: latex
[18:24:01][Commander] Building root file: %WS1%/pruebatex/main.tex
[18:24:01][Builder] Build root file %WS1%/pruebatex/main.tex
[18:24:01][Builder] outDir: %WS1%/pruebatex .
[18:24:01][Builder] Preparing to run recipe: latexmk.
[18:24:01][Builder] Prepared 1 tools.
[18:24:01][Builder] Recipe step 1 The command is latexmk:["-synctex=1","-interaction=nonstopmode","-file-line-error","-pdf","-outdir=%WS1%/pruebatex","%WS1%/pruebatex/main"].
[18:24:01][Builder] env: {}
[18:24:01][Builder] root: %WS1%/pruebatex/main.tex
[18:24:01][Builder] cwd: %WS1%/pruebatex
[18:24:01][Builder] LaTeX build process spawned with PID 21081.
[18:24:01][DupLabel] Checking for duplicate labels: %WS1%/pruebatex/main.tex .
[18:24:01][Cacher] Updated elements of %WS1%/pruebatex/main.tex .
[18:24:01][Cacher] Updated bibs of %WS1%/pruebatex/main.tex .
[18:24:01][Cacher] Cached %WS1%/pruebatex/main.tex .
[18:24:01][Structure] Structure force updated with 1 root sections for %WS1%/pruebatex/main.tex .
[18:24:02][Cacher][Watcher] "change" emitted on %WS1%/pruebatex/main.pdf after polling for 250 ms.
[18:24:03][Cacher][Watcher] "change" emitted on %WS1%/pruebatex/main.pdf after polling for 251 ms.
[18:24:03][Parser][TexLog] Logged 0 messages.
[18:24:03][Builder] Finished a step in recipe with PID 21081.
[18:24:03][Builder] Successfully built %WS1%/pruebatex/main.tex .
[18:24:03][Viewer] Call refreshExistingViewer: "%WS1%/pruebatex/main.tex" .
[18:24:03][Viewer] Refresh PDF viewer: undefined
[18:24:03][Cacher] Parsing .fls %WS1%/pruebatex/main.fls .
[18:24:03][Cacher] Found .aux %WS1%/pruebatex/main.tex from .fls %WS1%/pruebatex/main.fls , parsing.
[18:24:03][Cacher] Parsed .aux %WS1%/pruebatex/main.tex .
[18:24:03][Cacher] Parsed .fls %WS1%/pruebatex/main.fls .
[18:24:03][Cacher][Watcher] "change" emitted on %WS1%/pruebatex/main.pdf after polling for 251 ms.
```

### Developer Tools Console
> _Please paste the whole log messages below, not parts of ones. This console logs can sometimes be very important in many cases. To access the log, click `Help` -> `Toggle Developer Tools` -> `Console`._
```
  ERR Cannot read properties of undefined (reading 'state'): TypeError: Cannot read properties of undefined (reading 'state')
	at PdfViewerPanelSerializer.deserializeWebviewPanel (/config/extensions/james-yu.latex-workshop-9.7.0-universal/out/src/components/viewerlib/pdfviewerpanel.js:60:98)
	at async r.$deserializeWebviewPanel (/app/code-server/lib/vscode/out/vs/workbench/api/node/extensionHostProcess.js:98:176120)
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope text.tex.
Old grammar file: vscode-remote://mydomain.example.com/app/code-server/lib/vscode/extensions/latex/syntaxes/TeX.tmLanguage.json.
New grammar file: vscode-remote://mydomain.example.com/config/extensions/james-yu.latex-workshop-9.7.0-universal/syntax/TeX.tmLanguage.json
register @ TMScopeRegistry.ts:45
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope text.tex.latex.
Old grammar file: vscode-remote://mydomain.example.com/app/code-server/lib/vscode/extensions/latex/syntaxes/LaTeX.tmLanguage.json.
New grammar file: vscode-remote://mydomain.example.com/config/extensions/james-yu.latex-workshop-9.7.0-universal/syntax/LaTeX.tmLanguage.json
register @ TMScopeRegistry.ts:45
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope text.bibtex.
Old grammar file: vscode-remote://mydomain.example.com/app/code-server/lib/vscode/extensions/latex/syntaxes/Bibtex.tmLanguage.json.
New grammar file: vscode-remote://mydomain.example.com/config/extensions/james-yu.latex-workshop-9.7.0-universal/syntax/Bibtex.tmLanguage.json
register @ TMScopeRegistry.ts:45
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope text.tex.markdown_latex_combined.
Old grammar file: vscode-remote://mydomain.example.com/app/code-server/lib/vscode/extensions/latex/syntaxes/markdown-latex-combined.tmLanguage.json.
New grammar file: vscode-remote://mydomain.example.com/config/extensions/james-yu.latex-workshop-9.7.0-universal/syntax/markdown-latex-combined.tmLanguage.json
register @ TMScopeRegistry.ts:45
TMScopeRegistry.ts:45 Overwriting grammar scope name to file mapping for scope source.cpp.embedded.latex.
Old grammar file: vscode-remote://mydomain.example.com/app/code-server/lib/vscode/extensions/latex/syntaxes/cpp-grammar-bailout.tmLanguage.json.
New grammar file: vscode-remote://mydomain.example.com/config/extensions/james-yu.latex-workshop-9.7.0-universal/syntax/cpp-grammar-bailout.tmLanguage.json
register @ TMScopeRegistry.ts:45
log.ts:392  INFO Updating additional builtin extensions cache
viewer.js:13222 #bookmark1 is undefined.
translateElement @ viewer.js:13222
viewer.js:13222 #bookmark1_label is undefined.
translateElement @ viewer.js:13222
pdf.js:416 Warning: Setting up fake worker.
/proxy/build/pdf.worker.js:1          Failed to load resource: net::ERR_BLOCKED_BY_RESPONSE
/proxy/build/pdf.worker.js:1          Failed to load resource: the server responded with a status of 500 ()
viewer.js:2056 Ocurrió un error al cargar el PDF.

PDF.js v3.3.122 (build: 562045607)
Message: Setting up fake worker failed: "Cannot load script at: https://mydomain.example.com/proxy/build/pdf.worker.js".
_otherError @ viewer.js:2056
pdf.js:1960 Uncaught (in promise) Error: Setting up fake worker failed: "Cannot load script at: https://mydomain.example.com/proxy/build/pdf.worker.js".
    at pdf.js:1960:36
viewer.js:1909 The `PDFViewerApplication.open` signature was updated, please use an object instead.
open @ viewer.js:1909
viewer.js:2056 Ocurrió un error al cargar el PDF.

PDF.js v3.3.122 (build: 562045607)
Message: Setting up fake worker failed: "Cannot load script at: https://mydomain.example.com/proxy/build/pdf.worker.js".
_otherError @ viewer.js:2056
pdf.js:1960 Uncaught (in promise) Error: Setting up fake worker failed: "Cannot load script at: https://mydomain.example.com/proxy/build/pdf.worker.js".
    at pdf.js:1960:36
```

## Anything Else?
- The developer logs above are from the Chrome developer tools as I am using code-server instead of the desktop client.

**Comment by James-Yu** [comment]
We don’t provide official support to code-server. There are a few work arounds in the past issues that may help you resolve your problem.

**Comment by Pascal-D** [comment]
i found the problem why code-server is not working with latex-workshop and fixed it in my fork.

**Comment by golivax** [comment]
Is there an ETA for a new release that includes the fix from #3841?

**Comment by James-Yu** [comment]
After other PRs merged.
